proxy/request_handler.go

```diff
@@ -347,11 +347,14 @@ func (d *RequestHandler) InitializeAuthnSession(r *http.Request, rl *rule.Rule)
 			WithField("reason_id", "capture_groups_error").
 			Warn("Unable to capture the groups for the MatchContext")
 	} else {
+		for _, k := range d.c.ProxyAllowedClientHeaders() {
+			session.Header.Add(k, r.Header.Get(k))
+		}
+
 		session.MatchContext = authn.MatchContext{
 			RegexpCaptureGroups: values,
 			URL:                 r.URL,
 			Method:              r.Method,
-			Header:              r.Header,
```

Comment:
Is the purpose of this PR to reduce the headers to only the ones allowed in `d.c.ProxyAllowedClientHeaders`? If so, this would be a significant breaking change. Is there a particular reason for doing this?